File: work/python3/array/27_RemoveElement.py
#
# @lc app=leetcode id=27 lang=python3
#
# [27] Remove Element
#

# @lc code=start
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        last_uncheck=len(nums)-1
        current_check=0
        k=0
        while current_check <= last_uncheck:
            if nums[current_check] == val:
                nums[current_check],nums[last_uncheck]=nums[last_uncheck],nums[current_check]
                last_uncheck=last_uncheck-1
            else:
                k=k+1
                current_check=current_check+1
        
        return k

        
# @lc code=end

File: work/typescript/array/27_RemoveElement.ts
/*
 * @lc app=leetcode id=27 lang=typescript
 *
 * [27] Remove Element
 */

// @lc code=start
function removeElement(nums: number[], val: number): number {
    let k:number=nums.length-1;
    let i:number=0;
    while(i<=k){
        if(nums[i]==val){
            [nums[i],nums[k]]=[nums[k],nums[i]];
            k--;
        }
        else{
            i++
        };
    };
    return k+1;
};
// @lc code=end

